harness/determined/common/experimental/project.py

```diff
@@ -0,0 +1,127 @@
+import json
+from typing import Any, Dict, List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+
+
+class Project:
+    """A class representing a Project object.
+
+    Attributes:
+        workspace_id: (int) The ID of the workspace this project belongs to.
+        id: (int) The ID of the project.
+        archived: (Mutable, bool) True if experiment is archived, else false.
+        description: (Mutable, str) The description of the project.
+        name: (Mutable, str) Human-friendly name of the project.
+        notes: (Mutable, List[Dict[str,str]) Notes about the project. As determined upstream,
+            each note is a dict with exactly the keys "name" and "contents".
+        n_active_experiments: (int) The number of active experiments in the project.
+        n_experiments: (Mutable, int) The number of experiments in the project.
+    """
+
+    def __init__(
+        self,
+        session: api.Session,
+        workspace_id: int,
+        project_id: int,
+    ):
+        """Create a Project object.
+
+        Arguments:
+            session: The session to use for API calls.
+            workspace_id: ID of the workspace this project belongs to.
+            project_id: ID of the project.
+        """
+        self._session = session
+        self.workspace_id = workspace_id
+        self.id = project_id
+
+        # These properties may be mutable and will be set by _hydrate()
+        self.archived: Optional[bool] = None
+        self.description: Optional[str] = None
+        self.n_active_experiments: Optional[int] = None
+        self.n_experiments: Optional[int] = None
```

Comment:
Ooh, that's an _excellent_ idea. Totally. I agree that `n_experiments` etc. is weird, and this is a way better way to get the same information (should someone actually care).

Reply:
First took out `n_experiments` etc. and then put them back because those attributes are one of the only attributes shown in the CLI. Maybe that means somebody cares.